### ISSUE TITLE
NS3000: False positive when getting indexer value

### ISSUE THREAD

**Issue by reduckted** [nsubstitute/NSubstitute.Analyzers#218]
When using `CallInfo.ArgAt<T>` to get the value set into an indexer, the NS3000 diagnostic is produced even though the code runs without issues.

```cs
public interface IClass {
    int this[string name] { get; set; }
}

public class Test {
    public void Indexer() {
        IClass target = Substitute.For<IClass>();

        target.When((x) => x["foo"] = 42).Do(
            (info) => {
                string key = info.ArgAt<string>(0);
                int value = info.ArgAt<int>(1);
                //          ~~~~~~~~~~~~~~~~~~ There is no argument at position 1
            }
        );
    }
}
```
